CHANGELOG.md

```diff
@@ -21,6 +21,7 @@
 ### Changed
 
 - Major improvements to Flow types for Core Components ([499c195](https://github.com/facebook/react-native/commit/499c195), [fbc5a4f](https://github.com/facebook/react-native/commit/fbc5a4f), [f9050e0](https://github.com/facebook/react-native/commit/f9050e0), [6476151](https://github.com/facebook/react-native/commit/6476151), [c03fc40](https://github.com/facebook/react-native/commit/c03fc40), [69213ee](https://github.com/facebook/react-native/commit/69213ee), [136dfc8](https://github.com/facebook/react-native/commit/136dfc8), [3c0211b](https://github.com/facebook/react-native/commit/3c0211b), [c127000](https://github.com/facebook/react-native/commit/c127000), [636e146](https://github.com/facebook/react-native/commit/636e146), [6fa997d](https://github.com/facebook/react-native/commit/6fa997d), [35a65cd](https://github.com/facebook/react-native/commit/35a65cd), [7927497](https://github.com/facebook/react-native/commit/7927497), [45c5183](https://github.com/facebook/react-native/commit/45c5183), [a97d104](https://github.com/facebook/react-native/commit/a97d104), [fb4825a](https://github.com/facebook/react-native/commit/fb4825a), [84c5416](https://github.com/facebook/react-native/commit/84c5416), [3649a50](https://github.com/facebook/react-native/commit/3649a50) by [@mottox2](https://github.com/mottox2), [@saitoxu](https://github.com/saitoxu), [@RSNara](https://github.com/RSNara), [@watanabeyu](https://github.com/watanabeyu), [@Tnarita0000](https://github.com/Tnarita0000), [@exced](https://github.com/exced), [@nd-02110114](https://github.com/nd-02110114), [@flowkraD](https://github.com/flowkraD))
+- Components were converted to ES6 classes ([ScrollView](https://github.com/facebook/react-native/commit/221e2fe4095bc9ae15878725bdac4071d53e61f5)) by [@thymikee](https://github.com/thymikee). **Breaking Change**: Public methods of these components are no longer bound to their component instance. This was a behaviour of `createReactClass`, that we decided to not preserve when switching over to ES6 classes. 
```

Comment:
Should this be in a separate Breaking Changes section? I imagine it will be something people will need to be specifically aware of in their upgrades. 

Also, perhaps we can provide more information on the specific methods that have changed and an example of how their code will need to be changed.

Is there anything we can do to help people automatically detect these old calls in their codebase proactively without waiting for something to crash?

Really I just want to help people upgrade as easily as possible, providing any information / tools that helps make that happen.

Reply:
🤔 I think it's a good idea to have a breaking changes section. My only worry is that it'll never be all-encompassing/complete, unless we make an active effort to document all breaking changes to the codebase (and even then, we'll miss a few things). Also, unless we explicitly document that this section is not complete (which we could totally do), some people will assume that it is complete.

> Also, perhaps we can provide more information on the specific methods that have changed and an example of how their code will need to be changed.

I thought about this and I think it may be more succinct to say that all methods you can invoke on the component will no longer be bound to the component instance. But... I think it might be a good idea to list out the methods incase someone is CTRL+F'ing through the changelog. 

> Is there anything we can do to help people automatically detect these old calls in their codebase proactively without waiting for something to crash?